sycl/test/self-contained-handler-hpp.cpp

```diff
@@ -0,0 +1,17 @@
+// RUN: %clangxx -fsycl -fsyntax-only -Xclang -verify %s
+// expected-no-diagnostics
+//
+// The purpose of this test is to ensure that the header containing
+// sycl::handler class definition is self-contained, i.e. we can use handler
+// and no extra headers are needed.
```

Comment:
Can we have tighter integration with your generic tests for self-contained-headers? Something like not generating generic test for a header if it has a dedicated specific one (like this).

Reply:
So, I did a thing! See 0d071507f097441485e846636d1d853d5359f23d

Apparently, the whole `self-contained-headers` suite has been disabled since #14879 and it fails. Therefore, I will outline this commit into a separate PR, fixing all the issues we have along the way. But it is still published here as a backup and to get early feedback

Reply:
Outlined this upgrade into #16117